templates/repo/diff/box.tmpl

```diff
@@ -103,6 +103,11 @@
 									{{if $file.IsVendored}}
 										<span class="ui label gt-ml-3">{{$.locale.Tr "repo.diff.vendored"}}</span>
 									{{end}}
+									{{if and $file.Mode $file.OldMode}}
+										<span class="ui gt-ml-5 gt-mono">{{$file.Mode}} → {{$file.OldMode}}</span>
```

Comment:
I recommend extracting the modes into their own template, i.e.
```html
<div class="tooltip" data-content="{{if eq . 100755}}{{$.locale.Tr "repo.diff.executable_file"}}{{/*and so on*/}}{{end}}">{{.}}</div>
```
(or something similar)
and calling them then using i.e.
```js
{{template repo/diff/explain_filemode $file.Mode}}
```

Reply:
Hmm, why it's `{{$file.Mode}} → {{$file.OldMode}}` but not `OldMode -> Mode`?

----

About: "extracting the modes into their own template", is it worth to do so? This code is only used once, and it doesn't seem to be complex.

Reply:
Why extracting single translation?

Reply:
That's not what I meant.
I wanted to extract the single modes into a template to show a tooltip with what the new mode means.
As there are quite a few instances (three) where this tooltip would be needed, a template would be worth it.

Reply:
> what the given mode means

Do you want only the mode to be displayed like 755 or the file type as well (lile "symbolic link") ? What tooltips do you suggest besides executable file?

Reply:
Exactly these possible modes:
- `10755` - `Executable file`
- `10644` - `Non-executable file`
- `120000` - `Symbolic link`
- `040000` - `Directory`

Reply:
Don't think it's necessary to display IMO

Reply:
> Exactly these possible modes:
> 
>     * `10755` - `Executable file`
> 
>     * `10644` - `Non-executable file`
> 
>     * `120000` - `Symbolic link`
> 
>     * `040000` - `Directory`

Think it's better to check the first bit only. If it's odd then it's an executable

Reply:
Sorry, just can't make this extractable and still translatable. Maybe should not descrive file mode at all.